haystack/__init__.py

```diff
@@ -12,14 +12,15 @@
 __version__: str = str(metadata.version("farm-haystack"))
 
 
-# This configuration must be done before any import to apply to all submodules
+# Logging is not configured here so that developers using Haystack as a library can configure logging themselves.
+# If Haystack is used as an application, we suggest to use the following configuration:
+# import logging
+# logging.basicConfig(
+#     format="%(levelname)s - %(name)s -  %(message)s", datefmt="%m/%d/%Y %H:%M:%S", level=logging.WARNING
+# )
+# logging.getLogger("haystack").setLevel(logging.INFO)
```

Comment:
Not too important but I don't think we need this comment, or at least not a comment so detailed. Something like:

```python
# Logging not configured on purpose, see <link to issue>
```
should be sufficient.

Reply:
Alright, I shortened that comment. 👍